opr/assetList.go

```diff
@@ -44,10 +45,22 @@ func (o OraclePriceRecordAssetList) List(version uint8) []Token {
 
 // from https://github.com/iancoleman/orderedmap/blob/master/orderedmap.go#L310
 func (o OraclePriceRecordAssetList) MarshalJSON() ([]byte, error) {
+	if _, ok := o["version"]; !ok {
+		return nil, fmt.Errorf("marshaling json must be called through a safe function")
+	}
+	version := int(o["version"])
+	delete(o, "version")
```

Comment:
So in v1 we use the `json.Marshal(opr)`, but there is a problem. In the codebase, I switch everything to PEG. Rather than putting if statements everywhere on when to use `PNT` and when to use `PEG`, I thought it would be easier to use `PEG` everywhere. When we go to marshal the content (which is signed), we switch `PEG->PNT`. So the only place we ever see `PNT` is in the marshaled content. No code will ever use `PNT`, as when we unmarshal, we again replace `PNT` with `PEG` in the map.


The `version` thing is for 2 reasons. Calling `json.Marshal` is inherently unsafe, as now we have this rename business we ALWAYS need to address when marshaling content. The hashmap is already a custom marshal function that makes sure to marshal exactly the right assets, and nothing more.

The annoying part is the hashmap for all the currencies has no access to the OPR it is apart of, meaning it has no idea what `version` it is for. Because this is a `json.Marshal` override function, we cannot adjust the method definition without having to write our own Json encoding library (no thanks...). So I use an index in the hashmap as an indicator of which version to marshal for. It was that or creating a new structure, but creating a new structure would require touching almost every file (a lot touch the assets). In the protobuf rename, I am moving to use accessors vs using the map directly, partly because of this reason. Dealing with the raw map limits how I can add context to it (like a version number).